### PR TITLE
Hide map's overlays in 3d

### LINKED ISSUE #2783: Overlays are not well positionned
Measure overlays .

### PR DESCRIPTION
**Pull request by oterral**
Alternative to #2861
Fix #2783
[Test](https://mf-geoadmin3.dev.bgdi.ch/teo_overlays3d_hide/?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&dev3d=true&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bafu.wrz-wildruhezonen_portal,ch.swisstopo.swisstlm3d-wanderwege,KML%7C%7Chttps:%2F%2Fpublic.dev.bgdi.ch%2FcfG9RDtiR7W54Kg7RWwotA&layers_visibility=false,false,false,false,true&layers_timestamp=18641231,,,,&lon=7.25159&lat=46.62316&elevation=12260&heading=328.830&pitch=-43.38)

### CODE REVIEW

**Comment by gjn** [comment]
@oterral testlink does not work for me (I see nothing in 3D, not even background layer). Same behavioiur when checking out branch and compile myself.

**Comment by oterral** [comment]
I've currently no tiles on another branch too so it's not this PR in particular. I rebase in case.

**Comment by oterral** [comment]
Rebase redeployed @gjn

**Comment by gjn** [comment]
It also hides the context-popup link (crosshair=true parameter) in 3D. This should not be the case. Maybe there are others.

**Comment by oterral** [comment]
It's normal if it's an overlay we hide it on 3d because they are not well-positioned.
The question is should we display the crosshair as a vector layer ?

**Comment by gjn** [comment]
hmm. The icon we show when selecting a search result is not an overlay? It's the same icon as crosshair icon. So both should be shown, IMO. Measurements icons are a different animal.

**Comment by oterral** [comment]
We have transformed the search marker in a vector layer (old FeatureOverlay). I guess the crosshair is still an Overlay (= html elements).

**Comment by oterral** [comment]
So i think we could/should use a vector layer for the crosshair.

**Comment by gjn** [comment]
So it's probably best to transform crosshair to vector layer.

**Comment by oterral** [comment]
jenkins test

**Comment by oterral** [comment]
Pr update and rebased , can we merge @gjn?

File: src/components/map/MapDirective.js [hunk 1]
```diff
@@ -156,6 +156,43 @@ goog.require('ga_styles_service');
...
+                var val = evt.target.getPosition();
+                if (val) {
+                  item.set('realPosition', val);
+                  item.setPosition();
+                }
+              }));
+            };
+
+            // Watch when 3d is enabled to show/hide overlays
+            scope.$watch(function() {
```
Comment (gjn) [inline src/components/map/MapDirective.js]: Is this triggered on every digest cycle?

Comment (oterral) [inline src/components/map/MapDirective.js]: yes

Comment (gjn) [inline src/components/map/MapDirective.js]: It is, but it might be considered cheap.

Comment (oterral) [inline src/components/map/MapDirective.js]: Yes. we have this watch on multiple places maybe we should broadcast an event when 3d is active/deactive.

Comment (gjn) [inline src/components/map/MapDirective.js]: Yes, this might be worthwile.

**Pull request merged**